airflow/executors/base_executor.py

```diff
@@ -459,10 +466,14 @@ def success(self, key: TaskInstanceKey, info=None) -> None:
                 component="BaseExecutor",
                 parent_sc=gen_context(trace_id=trace_id, span_id=span_id),
             ) as span:
-                span.set_attribute("dag_id", key.dag_id)
-                span.set_attribute("run_id", key.run_id)
-                span.set_attribute("task_id", key.task_id)
-                span.set_attribute("try_number", key.try_number)
+                span.set_attributes(
+                    {
+                        "dag_id": key.dag_id,
+                        "run_id": key.run_id,
+                        "task_id": key.task_id,
+                        "try_number": key.try_number,
+                    }
+                )
```

Comment:
i think @ashb suggested something like, on TaskInstanceKey we could do something like

```
class TIKey:
    ...
    __span_attrs__ = ("dag_id", "run_id", "task_id", "try_number")
```

and then we could do something like 

```
span.add_ti_key_attrs(ti_key=ti_key)
```

just a one-liner.

wdyt?

Reply:
Yeah, for now I am doing the easy-thing i.e. not changing logic much -- only doing the smallest iteration. Will think about it in the follow-ups